Plugins/Data/NWScript/inc_array.nss

```diff
@@ -0,0 +1,485 @@
+void Array_Resize(string tag, int size, object obj=OBJECT_INVALID)
+{
+    if (obj == OBJECT_INVALID)
+        obj = GetModule();
+    ExecuteStatement("delete from "+GetTableName(tag, obj)+" where ind >= " + IntToString(size), obj);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+void Array_Shuffle(string tag, object obj=OBJECT_INVALID)
+{
+    if (obj == OBJECT_INVALID)
+        obj = GetModule();
+
+    string table = GetTableName(tag, obj);
+    ExecuteStatement("CREATE TABLE " +table+ "_temp AS SELECT ROW_NUMBER() OVER(ORDER BY RANDOM())-1, value FROM " +table, obj);
+    ExecuteStatement("DELETE FROM " +table , obj);
+    ExecuteStatement("INSERT INTO " +table+ " SELECT * FROM " +table+ "_temp", obj);
+    ExecuteStatement("DROP TABLE " +table+ "_TEMP", obj);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+int Array_Size(string tag, object obj=OBJECT_INVALID)
+{
+    return GetRowCount(tag, obj);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Sort the array by value according to 'direciton' (ASC or DESC)
+// Note that this is a lexical sort, so sorting an array of ints or floats will have
+// odd results
+void Array_Sort(string tag, string direction, int type, object obj=OBJECT_INVALID)
+{
+    string table = GetTableName(tag, obj);
+    if (obj == OBJECT_INVALID)
+        obj = GetModule();
+
+    // default orderBy for strings.
+    string orderBy = "ORDER BY value " + direction;
+    switch(type) {
+        case TYPE_INTEGER:
+            orderBy = "ORDER BY CAST(value AS INTEGER)" + direction;
+            break;
+        case TYPE_FLOAT:
+            orderBy = "ORDER BY CAST(value AS DECIMAL)" + direction;
+            break;
+    }
+    ExecuteStatement("CREATE TABLE " +table+  "_temp AS SELECT ROW_NUMBER() OVER(" + orderBy + ")-1, value FROM " +table, obj);
+    ExecuteStatement("DELETE FROM " +table, obj);
+    ExecuteStatement("INSERT INTO " +table+ " SELECT * FROM " +table+ "_temp", obj);
+    ExecuteStatement("DROP TABLE " +table+ "_temp", obj);
+}
+
+void Array_SortAscending(string tag, int type=TYPE_STRING, object obj=OBJECT_INVALID)
+{
+    Array_Sort(tag, "ASC", type, obj);
+}
+
+void Array_SortDescending(string tag, int type=TYPE_STRING, object obj=OBJECT_INVALID)
+{
+    Array_Sort(tag, "DESC", type, obj);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Set the value of array index 'index' to a 'element'
+// This will quietly eat values if index > array size
+void Array_Set_Str(string tag, int index, string element, object obj=OBJECT_INVALID)
+{
+     if (obj == OBJECT_INVALID)
+        obj = GetModule();
+    int rows = GetRowCount(tag, obj);
+    if (index < rows) {
+        ExecuteStatement("UPDATE "+GetTableName(tag, obj)+" SET value = '"+element+"' WHERE ind = "+IntToString(index), obj);
```

Comment:
It would be better to use binds here so there's no injection vulnerability (and in other similar places)